src/api/app/services/package_controller_service/rebuild_trigger.rb

```diff
@@ -0,0 +1,32 @@
+module PackageControllerService
+  class RebuildTrigger
+    def initialize(options = {})
+      @package = options[:package]
+      @project = options[:project]
+      @params = options[:params]
+    end
+
+    def rebuild?
+      @package.rebuild(@params.slice(*allowed_params).permit!.to_h)
+    end
+
+    def fetch_policy
+      return @project if @project != @package.project
```

Comment:
Please also "port" the documentation why we do this. It's very non-obvious...

Reply:
fixed